### PR TITLE
398/V2 Footer component and styling

### LINKED ISSUE #398: Footer component and styling
### Expected Behavior

The footer should match the new redesign.

### Acceptance Criteria

- Component should replace existing component, Footer.tsx
- Component should be self contained and have a corresponding storybook file 
- Styling should match the designs in Figma and use the matching colour variables outline in the tailwind config file
- Component should be responsive and should match layout designs for mobile and desktop (see Figma file)
- all links external and internal should work correctly
- Any e2e test that already exist should pass and new one should be added as necessary 

Image

Image

Link for Figma file is [here](https://www.figma.com/design/TAy2JI8BTACwaMjbuRRsak/Kyndly?node-id=376-2681&t=n8zrVLMPAhxd7Xdt-1})

### Current Behavior

n/a

### Any other notes

_No response_

### Assignment

This issue is free for anyone to take

### PR DESCRIPTION
**Pull request by OlenaReukova**
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review on my code
- [ ] I have made corresponding changes to the documentation (if any were needed)
- [ ] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and previously existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description

**Relates #398 <u>OR</u> Closes #398**  
Include here a short description of what behaviour this Pull Request introduces.
Outline any new dependencies introduced.

### Files changed

Include one to two lines on the necessary changes to each file.

### UI changes

If this PR includes changes to the UI include a **screengrab** of the alterations in review.

### Changes to Documentation

Include an outline of why Documentation needed to be updated with this PR

# Tests

Include a description of any tests added - if none have been added explain why.

### CODE REVIEW

**Comment by OlenaReukova** [comment]
@nichgalzin Hi.
**What I have done so far:**

- Added social icons.
- Created a new Footer component.
- Imported the new Footer component into the page and layout.
- Fully responsive.

**What I did not do:**

- I did not add the configuration settings from Tailwind CSS because the updated version has not yet been merged.
- Cypress Test
- Remove the previous Footer component.
- Add actual links to all elements of the footer, as, according to a conversation with the UI/UX design team, they will be provided at a later stage of the design development.

**Comment by nichgalzin** [comment]
Hey Olena, 
   I think go ahead and delete the old footer, I think it's a good idea in this footer to replace and remove anything that is not used anymore and just make sure to check all the places it's being used.
 As for cypress tests, lets at least make sure that we're not making any existing tests fail and that we create a storybook file for any new xu/ui components we create. Thanks!

File: components/footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+};
+
+const Footer: React.FC<FooterProps> = ({
+  companyInfo = {
+    name: 'KYNDLY CIC',
+    registrationNumber: '16077388',
+    address: '7 Bell Yard, London, WC2A 2JR',
+  },
+  socialLinks = {
+    facebook: 'https://facebook.com',
```
Comment (nichgalzin) [inline components/footer/Footer.tsx]: Let's ask Katia for the actual links to the Trafalgar girls social media pages and use them instead?

File: components/icons/ThreadsIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <svg
+      viewBox='0 0 24 24'
+      fill='none'
+      xmlns='http://www.w3.org/2000/svg'
+      className={className}
+    >
+      <g clipPath='url(#clip0)'>
+        <path
+          d='M18.75 7.61294C17.7994 5.14169 15.8034 3.33325 12 3.33325C6 3.33325 4.5 7.83325 4.5 12.3333C4.5 16.8333 6 21.3333 12 21.3333C16.5 21.3333 18.75 18.3333 18.75 16.0833C18.75 10.0833 9 10.0833 9 14.5833C9 18.3333 15.75 18.3333 15.75 12.3333C15.75 7.08325 10.5 7.08325 9 9.33325'
```
Comment (nichgalzin) [inline components/icons/ThreadsIcon.tsx]: This looks like it needs to be formatted?

File: components/icons/socialIcons.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export { default as FacebookIcon } from './FacebookIcon';
```
Comment (nichgalzin) [inline components/icons/socialIcons.ts]: We haven't been using this convention of have files dedicated to exporting. I'm not opposed to it but if this case can you make a directory called something like social-icons and then have all the component files and this in there?

File: components/footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  socialLinks?: {
+    facebook?: string;
+    instagram?: string;
+    threads?: string;
+  };
+};
+
+const Footer: React.FC<FooterProps> = ({
+  companyInfo = {
+    name: 'KYNDLY CIC',
```
Comment (nichgalzin) [inline components/footer/Footer.tsx]: Passing the hard coded values as props is fine, but I think we should not set the values as defaults here. Lets make a separate file with these objects, we should have a directory called constants with these sorts of files in them, we have a few other files like this that we need to move there, so go ahead and set that up for us.

**Review by nichgalzin — CHANGES_REQUESTED** [review]
Hi Oleana, great work this is looking good, just have a few notes:
- it seems like one the E2E tests is failing can you have a look at it and see what's wrong?
- the bg-color property is not working as you haven't pulled the latest changes that have now been merge into main. Can you make sure this branch is up to date before we merge it?
-On the desktop version the padding needs a bit of work, make sure there are even spacing of both sides, this it what it looks like to me.
Screenshot 2025-03-03 at 16 28 24

Thanks again for your work!
_____________________________
@nichgalzin 
updated info: 1) The Cypress tests are failing because cy.click() is being called on multiple elements instead of a single one.

Current errors in 'Anonymous Navigation':
✅ Can see recently added items (2188ms) - passed
❌ 1) Can browse items
❌ 2) Can see about page
❌ 3) Add item redirects to login
❌ 4) Message redirects to login
❌ 5) Profile redirects to login"
2) I merged the updated dev branch into the current branch, and now we don't have an issue with the bg-color property.
3) I fixed the alignment of the footer section by using flex and justify-between. Now, it looks centred with accurate right and left padding.

File: components/footer/Footer.tsx (hunk 4 above)
Comment (OlenaReukova) [reply components/footer/Footer.tsx]: 1. Moved hardcoded footer constants to 'constants/footer.ts' for better reusability.
2. Extracted 'FooterProps' type to 'types/footer.ts' to keep concerns separate.
3. Updated 'Footer.tsx' to import constants and types from new locations.
This refactor improves code organisation, makes updates and maintain easier.

File: components/footer/Footer.tsx (hunk 1 above)
Comment (OlenaReukova) [reply components/footer/Footer.tsx]: updated actual social media links

File: components/icons/ThreadsIcon.tsx (hunk 2 above)
Comment (OlenaReukova) [reply components/icons/ThreadsIcon.tsx]: refactored the paths for social media icons -done

File: components/icons/socialIcons.ts (hunk 3 above)
Comment (OlenaReukova) [reply components/icons/socialIcons.ts]: created social-icon folder and moved icons inside to keep all component files organised

**Review by nichgalzin — CHANGES_REQUESTED** [review]
A few notes:
- Mobile version of this looks great but is missing the links for the "privacy policy" and and "term and conditions"
- The spacing and padding  looks a bit strange when this is one a tablet sized device:
Screenshot 2025-03-24 at 11 07 36
- On desktop the padding need to have 100px gutter on either side of the text (let me know if this is not clear)
- We don't want the footer on the conversation page, but it currently is there

**Comment by OlenaReukova** [comment]
> A few notes:
> 
> * Mobile version of this looks great but is missing the links for the "privacy policy" and and "term and conditions"
> * The spacing and padding  looks a bit strange when this is one a tablet sized device:
>   Screenshot 2025-03-24 at 11 07 36
> * On desktop the padding need to have 100px gutter on either side of the text (let me know if this is not clear)
> * We don't want the footer on the conversation page, but it currently is there

_______________
@nichgalzin, thank you for your time and the code review, I am starting to fix it now.
1) the links for the "privacy policy" and "terms and conditions" will be provided by the UI/UX designers in the last stage. 
These pages are not in the prior list for them now.

2) Fixed the tablet's responsiveness - 4 columns.

3) On the desktop the padding is 100px gutter on either side of the text. 

4) Added DynamicFooter logic to exclude the Footer component from /conversations.

**Review by nichgalzin — APPROVED** [review]
Great work Olena. This all looks great.
